rc/lsp.kak

```diff
@@ -342,9 +342,39 @@ have_kakoune_feature_filtertext = ${kak_opt_lsp_have_kakoune_feature_filtertext}
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }}
 
+declare-option -hidden str lsp_completion_selection ""
+
 define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
+    set-option window lsp_completion_selection %val{hook_param}
+    try %{
+        eval -draft %{
+            select %opt{lsp_completion_selection}
+        }
+    } catch %{
+        # Completion was cancelled, nothing was inserted
+        # Clear these so the following lines do nothing.
+        # This is to avoid runing them in the above try block, and silencing their errors
+        remove-hooks window lsp-completion-dismissed
+        set-option window lsp_completions_selected_item -1
+    }
     lsp-completion-item-resolve-request false
+    trigger-user-hook "LspCompletionDismissed"
+    remove-hooks window lsp-completion-dismissed
     set-option window lsp_completions_selected_item -1
+    set-option window lsp_completion_selection ""
+}
+
+# User: lsp-completion-dismissed <command>
+# Run <command> when the completion menu is closed.
+# The inserted range is available as %opt{lsp_completion_selection} in the format expected by `select`.
+define-command -hidden lsp-completion-on-dismiss -params 1 %{
+    hook -once -group lsp-completion-dismissed window User "^LspCompletionDismissed$" %arg{1}
```

Comment:
The regex anchors ^ and $ are not necessary because `:doc hooks` says:

> filtering_regex must match the entire parameter string in order for the commands to be executed.